### ISSUE TITLE
Cannot read property `autoHideDropdown` of undefined

### ISSUE THREAD

**Issue by andrei-gheorghiu** [vickramravichandran/angular-auto-complete#21]
When using this in a tabbed form, if going away from and back to a tab containing an angular-auto-complete instance, the above error pops up. 

    TypeError: Cannot read property 'autoHideDropdown' of undefined
        at i.autoHide (dll.vendor.js:1)
        at dll.vendor.js:1
        at Object.b [as forEach] (dll.vendor.js:1)
        at Object.hideAllInactive (dll.vendor.js:1)
        at a (dll.vendor.js:1)
        at dll.vendor.js:1
        at c.$eval (dll.vendor.js:1)
        at c.$digest (dll.vendor.js:1)
        at c.$apply (dll.vendor.js:1)
        at dll.vendor.js:1

Ideally, the lib should provide a hook to delete any and all listeners placed on window object to allow removing them.

I pushed a quickfix on a fork and I'll make a pull-request. 

I simply replaced:

    if (that.options.autoHideDropdown) {
      ...

with

    if (that.options && that.options.autoHideDropdown) {
      ...

**Comment by vickramravichandran** [comment]
Is it possible to create a Fiddle/Codepen for this? I am curious why it does not fail in other places where `that.options` is accessed.

**Comment by andrei-gheorghiu** [comment]
Hi Vikram, thank you for the merge. 

Update on my situation: while trying to reproduce the issue in a fiddle I wasn't able to anymore. When looking deeper into it, I realized the issue happened on  v1.6.4,  not on v1.7.2. So, logically, I updated the project dependency to v1.7.3 but it looks like there are breaking changes between v1.6.4 and v1.7.3. 

I tried to update our code to work with with v1.7.3 but it looks like it's a no go: I can't seem to find a way to make the input show a label (it shows `[object Object]`), which in v1.6.4 was achieved by using: 

	...,
	renderItem: item => {
	  return {
	    value: item.name,
	    label: item.name
	  }
	},...

So, what I ended up doing was: reverted my fork to v1.6.4, applied the lil' fix and installed the package from my fork. If there's any way to achieve the above in latest version, i'd be happy to update. 

Thank you.

**Comment by vickramravichandran** [comment]
Yes, v1.7.0 had major refactoring. I added support for compilation (`ng-bind`) for the "label". But I did account for backward compatibility. Can you create a simple Plunkr using v1.6.4? I want to see what I broke...

I don't see why label would show `[object Object]`. See [example ](http://demo.vickram.me/angular-auto-complete/#!#custom-list-using-render-item-callback) using `renderItem` callback on the demo site. See "Custom List Using renderItem Callback".

**Comment by vickramravichandran** [comment]
I was able to reproduce the behavior. The fix after upgrading to 1.7.x is to set "label" to html
```javascript
	renderItem: item => {
	  return {
	    value: item.name,
	    label: "<span ng-bind='item.name'><span>"
	  }
	},
```
